docs/releases/2.2.txt

```diff
@@ -136,6 +136,13 @@ Database backends
 
 * Added result streaming for :meth:`.QuerySet.iterator` on SQLite.
 
+* Postgres backend will now check if the test DB exists before trying to create
+  it. If it does exist, a `DROP OWNED BY CURRENT_USER;` statement will be
+  issued to clear the DB before starting tests.
+
+  This resolves the problem where the user used to connect to the DB does not
+  have permission to create a db.
```

Comment:
Hey @funkybob. 

I'm not quite seeing the motivation for **us** _coding_ around this. Question: why should we not just say that the test user must have permissions to create the test db?

Reply:
Because for some users that is simply NOT an option.

Besides which, our current handling of the situation is, IMHO, a bit sloppy.

Reply:
> Because for some users that is simply NOT an option.

Really? 

Sorry! I've had plenty of coffee but I'm just not seeing it. What kind of test environment is going to not let you create a test DB? — or maybe create one for you and use `--keepdb`? (Saw the mailing list post but not sure why we can't make a demand on the test environment/invocation here?)

> Besides which, our current handling of the situation is, IMHO, a bit sloppy

Yes, that may be true. 🙂 

Is this worth a TRAC ticket with some background? (I'm just trying to get clear on the motivation/need.) 

Thanks! (Not least for the effort of helping me to understand!!)

Reply:
Because some people work in organisations where they are not allowed that level of control. I don't agree with it, but I _do_ encounter them on a somewhat regular basis.

The motivation is to remove the pain for the people I periodically encounter for whom this is frequently a painful issue to try to find a work-around for.

--keepdb isn't necessarily a solution, either.  It's optional for a reason (or possibly many).
Though I will try recommending that next time it crops up.